### PR TITLE
Bump btcec to v2, use integrated btcutil module

### PR DESCRIPTION
**Pull request by guggero**
Extracted out from https://github.com/lightningnetwork/lnd/pull/6263.

This PR switches `lnd` over to using the new `btcec/v2` and `btcutil` modules located in the `btcsuite/btcd` repo.

**All commits starting with `SQUASHME` will be squashed into a single commit before merging**. They are only pushed separately for easier review.

### CODE REVIEW

**Comment by Roasbeef** [comment]
Seeing this test failure uniformly: 
```
--- FAIL: TestLightningWallet (114.32s)
    --- FAIL: TestLightningWallet/btcwallet/neutrino:publish_transaction (15.92s)
        test_interface.go:1914: expected ErrDoubleSpend, got: <nil>
```

I think it might be due to: https://github.com/lightninglabs/neutrino/pull/240 ?

File: go.mod
Comment (Roasbeef) [inline go.mod]: If we remove the replace directives, then we don't need to modify this, and instead do it in another PR after allt he new tags are pushed.

**Comment by Roasbeef** [comment]
I think you may need to go into _each_ sub-module and to `go mod tidy`, so it can pick up other dependencies like the `psbt` package.

**Comment by guggero** [comment]
> I think it might be due to: https://github.com/lightninglabs/neutrino/pull/240 ?

Yes, you were right. It looks like that PR didn't account for reject messages correctly. I created a fix for it in https://github.com/lightninglabs/neutrino/pull/247 and downgraded this PR to a version just before https://github.com/lightninglabs/neutrino/pull/240 was merged.

**Comment by 3nprob** [comment]
Related: https://github.com/lightningnetwork/lnd/pull/5651

**Comment by Roasbeef** [comment]
Needs a rebase! We'll eventually update to that neutrino version before we tag the next major release, so fine w/ going with the downgraded version here.

**Comment by guggero** [comment]
Rebased.

**Comment by guggero** [comment]
Rebased again. @positiveblue PTAL.

**Comment by Roasbeef** [comment]
Modules pops up with this warning: 
> no required module provides package github.com/btcsuite/btcd/btcutil/gcs/builder; to add it:

I think it's missing a `go mod tidy` somewhere? But perhaps it's also just due to the set of intermediate commits.

**Review by Roasbeef — APPROVED** [review]
LGTM 🍭

Now that the PR uses the _prior_ version of neutrino, we're no longer blocked on that fix, allowing us to move forward here on the taproot front!

**Comment by guggero** [comment]
> Modules pops up with this warning:
> 
> > no required module provides package github.com/btcsuite/btcd/btcutil/gcs/builder; to add it:
> 
> I think it's missing a `go mod tidy` somewhere? But perhaps it's also just due to the set of intermediate commits.

Yeah, the intermediate commits (prefixed with `SQUASHME-`) won't build on their own. My plan is to squash them into a single commit before merging.

**Review by positiveblue — APPROVED** [review]
LGTM @guggero  thanks for splinting this in its own PR

The errors that Github actions show are only because some of the commits do not compile by itself but after squashing everything should pass (as it does in the last commit)

**Comment by guggero** [comment]
Thank you for the review, @positiveblue!

Going to squash the commits prefixed with `SQUASHME-` into a single one.
Adding this comment here to preserve the split on GitHub:

Parts:
 - 2d746f68: replace imports
 - 4008f0fd: use ecdsa.Signature
 - 849e33d1: remove btcec.S256()
 - b8f6ebbd: use v2 library correctly
 - fa80bca9: bump go modules

**Pull request merged**
